# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerUtils.h
# Commit: 244df402d6eb
# Full Hash: 244df402d6ebc15d6845c4144245511021408835
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2024-10-24 09:44:34
# Regressor Bug: 1131324
# File Overlap Count: 1
# Description:
#   Bug 1131324 - Expose ServiceWorkerContainer on WorkerNavigator. r=dom-worker-reviewers,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D213725
# ==============================================================================

diff -r 57f70ba5f27e -r 244df402d6eb dom/serviceworkers/ServiceWorkerUtils.h
--- a/dom/serviceworkers/ServiceWorkerUtils.h	Thu Oct 24 03:02:39 2024 +0000
+++ b/dom/serviceworkers/ServiceWorkerUtils.h	Thu Oct 24 03:02:40 2024 +0000
@@ -11,6 +11,7 @@
 #include "mozilla/dom/ServiceWorkerRegistrationDescriptor.h"
 #include "nsTArray.h"
 
+class nsIGlobalObject;
 class nsIURI;
 
 namespace mozilla {
@@ -51,9 +52,25 @@
 bool ServiceWorkerRegistrationDataIsValid(
     const ServiceWorkerRegistrationData& aData);
 
-void ServiceWorkerScopeAndScriptAreValid(const ClientInfo& aClientInfo,
-                                         nsIURI* aScopeURI, nsIURI* aScriptURI,
-                                         ErrorResult& aRv);
+// Performs key spec validation steps of
+// https://w3c.github.io/ServiceWorker/#start-register-algorithm and
+// https://w3c.github.io/ServiceWorker/#register-algorithm as well as CSP
+// validation corresponding to
+// https://w3c.github.io/webappsec-csp/#directive-worker-src.
+//
+// This is extracted out of ServiceWorkerContainer::Register because we validate
+// both in the content process as the site of the call, as well as in the parent
+// process in the ServiceWorkerManager.
+//
+// On worker threads, this will involve use of a syncloop until Bug 1901387 is
+// addressed, allowing us to call CheckMayLoad off main-thread (OMT).
+//
+// A global may be optionally provided for reporting purposes; this is desired
+// when this is used by ServiceWorkerContainer::Register but not necessary in
+// the parent process.
+void ServiceWorkerScopeAndScriptAreValid(
+    const ClientInfo& aClientInfo, nsIURI* aScopeURI, nsIURI* aScriptURI,
+    ErrorResult& aRv, nsIGlobalObject* aGlobalForReporting = nullptr);
 
 bool ServiceWorkersEnabled(JSContext* aCx, JSObject* aGlobal);
 